### PR TITLE
fix(codemirror): remove inline props

### PR DESCRIPTION
**Pull request by danilowoz**
CodeSandbox logoCodeSandbox logo&nbsp; Open in CodeSandbox <a href="https://codesandbox.io/p/github/codesandbox/sandpack/fix/extension-inline-prop">Web Editor</a> | <a href="https://codesandbox.io/p/vscode?owner=codesandbox&repo=sandpack&branch=fix/extension-inline-prop">VS Code</a>

Address https://github.com/reactjs/reactjs.org/issues/4686#issuecomment-1137402613

All `useEffect` which depends on inline default props (`extensions` and `extensionsKeymap`), were constantly being triggered unnecessarily due to these props always returning a new value every time that the parent rerender.  

@gaearon does it make sense? 

----

Note to me: apply these changes on `main` too.

### CODE REVIEW

**Comment by gaearon** [comment]
What if the user specifies an inline value? Like `extensions={[onLint]}`. I think you 
might need to diff them manually inside the effect instead.

**Review by gaearon — APPROVED** [review]

**Comment by danilowoz** [comment]
Indeed there is a significant different between them: 
Screenshot 2022-05-26 at 16 53 34

File: sandpack-react/src/utils/array.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export const shallowEqual = (a: any[], b: any[]): boolean => {
+  if (a.length !== b.length) return false;
+
+  return a.every((item, index) => item === b[index]);
```
Comment (gaearon) [inline sandpack-react/src/utils/array.ts:6]: Nit: this always traverses all the items. A regular for loop would be able to exit early on the first mismatches. Doesn't matter for small arrays though.

Comment (gaearon) [reply sandpack-react/src/utils/array.ts:6]: Actually wait, I'm wrong. I think it _would_ exit early. So nvm.

Comment (danilowoz) [reply sandpack-react/src/utils/array.ts]: If I'm doing this right, for-loop is still the fastest option compared to the `some` or `every` implementation: 
Screenshot 2022-05-26 at 17 33 08

Comment (danilowoz) [reply sandpack-react/src/utils/array.ts]: As far as I know, `some` or `every` exit as soon as a predicate match, but I couldn't find this in any documentation. If you have it, please let me know

Comment (gaearon) [reply sandpack-react/src/utils/array.ts]: loops ftw

Comment (sophiebits) [reply sandpack-react/src/utils/array.ts]: Yes, it does an early exit ("If _testResult_ is **false**, return **false**."). https://tc39.es/ecma262/#sec-array.prototype.every

**Pull request merged**